docs/build/apps/ingest-sdk/ingestion-pipeline-code.mdx

```diff
@@ -142,119 +144,121 @@ func (transformer *AppPaymentTransformer) Process(ctx context.Context, msg Messa
   return nil
 }
 
-type CaptiveCoreInboundAdapter struct {
-  TomlParams                ledgerbackend.CaptiveCoreTomlParams
+type LedgerMetadataInboundAdapter struct {
   processors                []Processor
   historyArchiveURLs        []string
-  coreConfigNetworkTemplate []byte
+  dataStoreConfig           datastore.DataStoreConfig
 }
 
-func (adapter *CaptiveCoreInboundAdapter) Subscribe(receiver Processor) {
+func (adapter *LedgerMetadataInboundAdapter) Subscribe(receiver Processor) {
   adapter.processors = append(adapter.processors, receiver)
 }
 
-func (adapter *CaptiveCoreInboundAdapter) Run(ctx context.Context) error {
-  // Setup captive core config to use the Pubnet network
-  captiveCoreToml, err := ledgerbackend.NewCaptiveCoreTomlFromData(adapter.coreConfigNetworkTemplate, adapter.TomlParams)
-  if err != nil {
-    return errors.Wrap(err, "error creating captive core toml")
-  }
-
-  captiveConfig := ledgerbackend.CaptiveCoreConfig{
-    BinaryPath:         adapter.TomlParams.CoreBinaryPath,
-    HistoryArchiveURLs: adapter.TomlParams.HistoryArchiveURLs,
-    Context:            ctx,
-    Toml:               captiveCoreToml,
-  }
-
-  // Create a new captive core backend, the origin of ingestion pipeline
-  captiveBackend, err := ledgerbackend.NewCaptive(captiveConfig)
-  if err != nil {
-    return errors.Wrap(err, "error creating captive core instance")
-  }
+func (adapter *LedgerMetadataInboundAdapter) Run(ctx context.Context) error {
 
-  // Create a client to the network's history archives
-  historyAchive, err := historyarchive.NewArchivePool(adapter.historyArchiveURLs, historyarchive.ArchiveOptions{
+  /////////////////////////////////////////////////////////////////
+  // TODO - make this easier for callers, move latest ledger fetch into cdp.ApplyLedgerMetadata()
```

Comment:
Erm, is it typical to leave TODO comments in developer docs, or did you mean to clean this up?

Reply:
ah, thanks, I need to remove, will reference to [dev ticket](https://github.com/stellar/go/issues/5495) instead